docking.js

```diff
@@ -2198,112 +2161,10 @@ export class DockManager {
                 return [0, 0];
             });
 
-        const {ControlsManager} = OverviewControls;
         // FIXME: https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2890
         // const { ControlsManagerLayout } = OverviewControls;
         const ControlsManagerLayout = this.overviewControls.layout_manager.constructor;
 
-        this._methodInjections.removeWithLabel(Labels.STARTUP_ANIMATION);
-        this._methodInjections.addWithLabel(Labels.STARTUP_ANIMATION,
-            Main.layoutManager.constructor.prototype, '_startupAnimationComplete',
-            originalMethod => {
-                originalMethod.call(Main.layoutManager);
-                this._methodInjections.removeWithLabel(Labels.STARTUP_ANIMATION);
-                this._signalsHandler.removeWithLabel(Labels.STARTUP_ANIMATION);
-            });
-
-        if (Main.layoutManager._startingUp && Main.layoutManager._waitLoaded) {
-            // Disable this on versions that will include:
-            //  https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2763
-            this._methodInjections.addWithLabel(Labels.STARTUP_ANIMATION,
-                Main.layoutManager.constructor.prototype,
-                '_prepareStartupAnimation', function (originalMethod, ...args) {
-                    /* eslint-disable no-invalid-this */
-                    const dockManager = DockManager.getDefault();
-                    const temporaryInjections = new Utils.InjectionsHandler(
-                        dockManager);
-
-                    const waitLoadedHandlingMonitors = (_, bgManager) => {
-                        return new Promise((resolve, reject) => {
-                            const connections = new Utils.GlobalSignalsHandler(
-                                dockManager);
-                            connections.add(bgManager, 'loaded', () => {
-                                connections.destroy();
-                                resolve();
-                            });
-
-                            connections.add(Utils.getMonitorManager(), 'monitors-changed', () => {
-                                connections.destroy();
-
-                                reject(new GLib.Error(Gio.IOErrorEnum,
-                                    Gio.IOErrorEnum.CANCELLED, 'Loading was cancelled'));
-                            });
-                        });
-                    };
```

Comment:
Ideally I'd be very happy to drop all this code, but one reason for that was that https://gitlab.gnome.org/GNOME/gnome-shell/-/merge_requests/2763 was still a thing at the time.

I've not tested anymore the script, but that was triggering the same root issue we had that was leading to a black/non-responsive screen input area.

So, have you tested with tricks like those and see if we're strong enough now?

Reply:
I don't remember all the details of those clipping bugs, only that it happened twice in the past. So we should investigate, but I am keen to build on a much smaller code base if any new fix is required. I vaguely recall finding upstream had the same bug, so that plus the fact that we no longer modify the upstream animation, should mean any fix required for the clipping issue could go upstream.

Reply:
Sorry for the delay...

The clipping bug does still happen upstream (no extensions), but only for logging into `Main.sessionMode.hasOverview == false` which is not the default. And the clipping gets corrected at the end of the login animation.

dash-to-dock master does not have the bug, but to test that you also need to fix the undefined `STARTUP_ANIMATION_TIME` issue or else there's no way to slow down animations.

dash-to-dock #2166 does have the bug, but it now corrects itself at the end of the login animation just like upstream does.

So we no longer need to carry the clipping fix here. The bug is now the same with or without extensions after this PR. It should be treated as [an upstream bug](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/6473) from now on.

Reply:
So I checked the script and the bug is back... mhmh

Reply:
Yeah that's kind of what I said in the last two paragraphs. I think it's OK. It recovers at the end of the animation and [the same bug](https://gitlab.gnome.org/GNOME/gnome-shell/-/issues/6473) occurs without extensions so any fix should go in upstream gnome-shell.

Reply:
Since it does recover at the end of the animation the bug shouldn't be visible at all when animations are disabled (like in the original VM case). But I forgot to test with animations disabled. I will need to do that.